spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/ChangelogIterator.java

```diff
@@ -147,19 +163,6 @@ private int[] generateIndicesForIdentifySameRow(Row row) {
     return indices;
   }
 
-  private Row[] createUpdateChangelog(
-      GenericRowWithSchema currentRow, GenericRowWithSchema nextRow) {
-    GenericInternalRow deletedRow = new GenericInternalRow(currentRow.values());
-    GenericInternalRow insertedRow = new GenericInternalRow(nextRow.values());
-
-    deletedRow.update(changeTypeIndex, UPDATE_BEFORE);
-    insertedRow.update(changeTypeIndex, UPDATE_AFTER);
-
-    return new Row[] {
-      RowFactory.create(deletedRow.values()), RowFactory.create(insertedRow.values())
-    };
-  }
-
   private boolean isCarryoverRecord(Row currentRow, Row nextRow) {
     for (int idx : indicesForIdentifySameRow) {
       if (!isColumnSame(currentRow, nextRow, idx)) {
```

Comment:
My 2 cents, it seems that modify this method as :
```java
  private boolean isCarryoverRecord(Row currentRow, Row nextRow) {
    for (int idx = 0; idx < currentRow.length(); idx++) {
      if (idx != this.changeTypeIndex) {
        if (!isColumnSame(currentRow, nextRow, idx)) {
          return false;
        }
      }
    }

    return true;
  }
```
allows to remove indicesForIdentifySameRow array

Reply:
To precompute would be faster than to check within the loop.  More details are in https://github.com/apache/iceberg/pull/6344/files/e813670402459901ed4f2b36cb24349145b705ac#r1060949312.